frontend/packages/dev-console/src/components/pipelines/PipelineDetailsPage.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { DetailsPage, DetailsPageProps } from '@console/internal/components/factory';
+import { Kebab, navFactory } from '@console/internal/components/utils';
+import { k8sGet, k8sList } from '@console/internal/module/k8s';
+import PipelinEnvironmentComponent from './PipelineEnvironment';
+import PipelineDetails from './PipelineDetails';
+import PipelineRuns from './PipelineRuns';
+import { triggerPipeline, rerunPipeline } from '../../utils/pipeline-actions';
+import { getLatestRun } from '../../utils/pipeline-augment';
+import { PipelineRunModel, PipelineModel } from '../../models';
+
+interface PipelineDetailsPageStates {
+  menuActions: Function[];
+}
+
+class PipelineDetailsPage extends React.Component<DetailsPageProps, PipelineDetailsPageStates> {
+  constructor(props) {
+    super(props);
+    this.state = { menuActions: [] };
+  }
+
+  componentDidMount() {
+    // eslint-disable-next-line promise/catch-or-return
+    k8sGet(PipelineModel, this.props.name, this.props.namespace).then((res) => {
```

Comment:
Error handling?

Reply:
I will create separate bugs for the team to address these.